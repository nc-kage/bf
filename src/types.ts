export type OrderType = [number, [number, number, number]];
export type OrdersType = [number, [number, number, number][]];

export type JsonType<P = null | string | number | boolean> = {
  [key: string]: P | JsonType<P> | JsonType<P>[];
};

export enum OrderSymbol {
  BTCUSD = 'tBTCUSD',
  ETHUSD = 'tETHUSD',
}

export enum OrderProposition {
  BID = 1,
  ASK = -1,
}

export enum MessageEvent {
  SUBSCRIBE = 'subscribe',
  SUBSCRIBED = 'subscribed',
  UNSUBSCRIBE = 'unsubscribe',
  UNSUBSCRIBED = 'unsubscribed',
}

export enum Channel {
  BOOK = 'book',
}
