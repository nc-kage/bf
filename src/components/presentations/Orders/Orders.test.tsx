import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Comp from '.';
import { OrderProposition } from '@/types';

jest.mock('@/components/containers/Order', () => (): null => null);

describe('App', () => {
  it('snapshot left', () => {
    const { asFragment } = render(
      <Comp
        orderPrices={[1, 2]}
        orderPosition={OrderProposition.BID}
        orientation="left"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot right', () => {
    const { asFragment } = render(
      <Comp
        orderPrices={[1, 2]}
        orderPosition={OrderProposition.BID}
        orientation="right"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
