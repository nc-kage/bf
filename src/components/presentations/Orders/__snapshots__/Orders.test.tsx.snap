// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot left 1`] = `
<DocumentFragment>
  <div
    class="Orders"
  >
    <div
      class="Orders__header Orders__header--left"
    >
      <span
        class="Orders__header__item"
      >
        Count
      </span>
      <span
        class="Orders__header__item"
      >
        Amount
      </span>
      <span
        class="Orders__header__item"
      >
        Price
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App snapshot right 1`] = `
<DocumentFragment>
  <div
    class="Orders"
  >
    <div
      class="Orders__header Orders__header--right"
    >
      <span
        class="Orders__header__item"
      >
        Price
      </span>
      <span
        class="Orders__header__item"
      >
        Amount
      </span>
      <span
        class="Orders__header__item"
      >
        Count
      </span>
    </div>
  </div>
</DocumentFragment>
`;
